tests/data/power_op_spacing.py

```diff
@@ -0,0 +1,65 @@
+def function(**kwargs):
+    t = a**2 + b**3
+    return t ** 2
+
+
+def function_replace_spaces(**kwargs):
+    t = a **2 + b** 3 + c ** 4
+
+
+def function_dont_replace_spaces():
+    {**a, **b, **c}
+
+
+a = 5**~4
+b = 5 ** f()
+c = -(5**2)
+d = 5 ** f["hi"]
+e = lazy(lambda **kwargs: 5)
+f = f() ** 5
+g = a.b**c.d
+h = 5 ** funcs.f()
+i = funcs.f() ** 5
+j = super().name ** 5
+k = [(2**idx, value) for idx, value in pairs]
+l = mod.weights_[0] == pytest.approx(0.95**100, abs=0.001)
+m = [([2**63], [1, 2**63])]
+n = count <= 10**5
+o = settings(max_examples=10**6)
+p = {(k, k**2): v**2 for k, v in pairs}
+q = [10**i for i in range(6)]
+
```

Comment:
A fun case to test might be `1.0.real ** 5.0.real`. I'd say that should not hug but I don't feel strongly if it's easier to implement the other way.

Reply:
Also could use some tests with float literals instead of int literals.

Reply:
I don't care either way but this feels like it goes against the allowance of attribute access in the rule stated above. We could stop hugging attribute access entirely but I'd argue it's more readable once you add other operators:
```python
f = bound = Decimal(10 ** field.max_digits - 1) / (10 ** field.decimal_places) 
g = a.b**c.d + factor * 2

# vs.

f = Decimal(10**field.max_digits - 1) / (10**field.decimal_places)
g = a.b ** c.d + factor * 2
```

For the record making this case not hug should be as simple as:
```diff
@@ -101,6 +101,10 @@ def hug_power_op(line: Line, features: Collection[Feature]) -> Iterator[Line]:
                 # of a comprehension.
                 return True
 
+            # Attribute access on numeric constants is also disallowed.
+            if current.type == token.NAME:
+                disallowed.add(token.NUMBER)
```